strapi/strapi
# replace axiosInstance with getFetchClient inside the email

### What does it do?

## Replace every instance of axiosInstance with the new getFetchClient in the email plugin

We have already implemented the new constructs useFetchClient and getFetchClient in[ this PR](https://github.com/strapi/strapi/pull/14759)

Now we need to replace every place where we use axiosInstance with the new constructs, test the possible regression errors and still provide the old implementation of the axiosInstance into the helper plugin with a message of the removal in v5

### Why is it needed?

Inside the Strapi project we have a lot of different implementations of the same code to use axios and its axiosInstance. In this branch we replace axiosInstance in the email plugin

### How to test it?

You need to use the new construct inside every axiosInstance call and check the possible regression errors

List of the API calls in the Email where we use the new construct and how to test them:
Here you can see a list of API calls changed with the instructions to test them:

 - fetch Email Settings GET, http://localhost:1337/email/settings from the Admin click on “Settings“, under the “GENERAL“, then click on "Configuration" under the "EMAIL PLUGIN" section, it will trigger the API call, as you can see in these screenshots https://www.loom.com/i/a18d8ec951604b15a4355603f3ed6898
https://www.loom.com/i/f115b3f51b0f4529ba7061117fd058fb
https://www.loom.com/i/71b85a1d2e724f39ab7ac15b54fed498
https://www.loom.com/i/ff04a2db33ec4462916bce10baee883c
https://www.loom.com/i/13eabe0e1fda465da52959be61354a0d
https://www.loom.com/i/db7f1727361545a38973ad74df7e92f1
https://www.loom.com/i/3e40248bda554cb5b6dd2dbf7f84bcda
https://www.loom.com/i/b748b98e198d447dbc675fa1afc73220

 - test Email Settings POST, http://localhost:1337/email/test from the Admin click on “Settings“, under “GENERAL“, then click on "Configuration" under the "EMAIL PLUGIN" section; fill the form to Test the email delivery with an email and submit clicking on "Submit test email", it will trigger the API call, as you can see in these screenshots https://www.loom.com/i/3f4c02ba40374174a50899d8d23a3757
https://www.loom.com/i/e9fd6207135c4438908ac107d861a4c7
https://www.loom.com/i/2faf4ead640f425e8d64851d7efbad2e
https://www.loom.com/i/e4a493dcee6a474d904c0cbeb986073e
https://www.loom.com/i/2cba4d514a4a4101a779d3cb366f4d73
https://www.loom.com/i/6b7d2f392c774c91819cacd2a81f28d8
https://www.loom.com/i/8f777a3cbc1f4cbc84c2298cc2c1c477
https://www.loom.com/i/41eb8c8eeecc46168cef0fac4eb575ab
https://www.loom.com/i/c0ea6cf07e3c49fcac2db16cacf76fe8
https://www.loom.com/i/2dde4c9b58834fe984726624b2026d02
(I always had, even with axiosInstance implementation a 500 timeout response from the API)
